### DISCUSSION TITLE
What scopes are required on gitea PAT?

### DISCUSSION THREAD

**Discussion by benoit-garcia** [renovatebot/renovate#22670] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.115.0

### If you're self-hosting Renovate, select which platform you are using.

Gitea

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Documentation contains information regarding scopes that are required on Gitea PAT.

### What you tried so far.

Create a PAT without selecting any scope, as none are required. Running renovate fails with a 403.

When using a token with all scopes, it works fine. I doubt all scopes are relevant; I think [the documentation](https://docs.renovatebot.com/modules/platform/gitea/) could be improved by indicated the scope to select.

### Relevant debug logs

<details><summary>Logs</summary>

```
 WARN: Host error (repository=<GITEA_USER>/<GIT_REPO>)
       "hostType": "git",
       "packageName": undefined,
       "err": {
         "task": {
           "commands": [
             "ls-remote",
             "--heads",
             "https://**redacted**@<GITEA_URL>/<GITEA_USER>/<GIT_REPO>.git"
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "message": "remote: Forbidden\nfatal: unable to access 'https://<GITEA_URL>/<GITEA_USER>/<GIT_REPO>.git/': The requested URL returned error: 403\n",
         "stack": "Error: remote: Forbidden\nfatal: unable to access 'https://<GITEA_URL>/<GITEA_USER>/<GIT_REPO>.git/': The requested URL returned error: 403\n\n    at Object.action (/opt/homebrew/Cellar/renovate/35.115.0/libexec/lib/node_modules/renovate/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:42:29)\n    at PluginStore.exec (/opt/homebrew/Cellar/renovate/35.115.0/libexec/lib/node_modules/renovate/node_modules/simple-git/src/lib/plugins/plugin-store.ts:29:29)\n    at /opt/homebrew/Cellar/renovate/35.115.0/libexec/lib/node_modules/renovate/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:127:42\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/opt/homebrew/Cellar/renovate/35.115.0/libexec/lib/node_modules/renovate/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:124:14)\n    at GitExecutorChain.<anonymous> (/opt/homebrew/Cellar/renovate/35.115.0/libexec/lib/node_modules/renovate/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:100:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/opt/homebrew/Cellar/renovate/35.115.0/libexec/lib/node_modules/renovate/node_modules/simple-git/dist/cjs/index.js:55:24)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)"
       }
```

</details>

**Comment by viceice** [comment]
🤷‍♂️ you need to figure it out and let us know. as we added Gitea support there were no scopes. so most existing setups are working with full token's.

**Reply by benoit-garcia** [reply]
Thanks for your reply.
For what it worth, it looks to work with the "repo" scope.

**Reply by viceice** [reply, marked as answer]
I think user scope is required when you don't configure username or gitAuthor.

required scopes:
- `repo`
- `read:user` and `read:email` or configure `gitAuthor` and `username`
- `read:packages` (when using Gitea packages)

**Reply by viceice** [reply]
@HonkingGoose can you add this to the docs? requires Gitea v1.19+

**Reply by HonkingGoose** [reply]
Here's my PR to update the docs:

- #22685
